Add plugin: Smart Link Alias

vpcano opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: [vpcano/obsidian-smart-link-alias](https://github.com/vpcano/obsidian-smart-link-alias)

## Release Checklist
- [ ] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[document.addEventListener('mousedown', () => {](https://github.com/vpcano/obsidian-smart-link-alias/blob/3a24c13804852962f5ef2c605cf8614750cca906/src/link-decorations.field.ts#L123-L128)
Events need to be registered, so they can be properly unloaded when the plugin is disabled. <https://docs.obsidian.md/Plugins/Events>

vpcano commented:
> [document.addEventListener('mousedown', () => {](https://github.com/vpcano/obsidian-smart-link-alias/blob/3a24c13804852962f5ef2c605cf8614750cca906/src/link-decorations.field.ts#L123-L128) Events need to be registered, so they can be properly unloaded when the plugin is disabled. https://docs.obsidian.md/Plugins/Events
> 
> Also, these event's won't run when in a pop out window, you will have to listen for the `window-open` event from Obsidian to handle that properly.

Hello. I've fixed both things and a bug with the links not being displayed on headers.

I've updated the [release](https://github.com/[vpcano/obsidian-smart-link-alias](https://github.com/vpcano/obsidian-smart-link-alias/releases/tag/1.0.0)/releases/tag/1.0.0) with the new changes.
